src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -811,374 +802,261 @@ MavlinkReceiver::handle_message_att_pos_mocap(mavlink_message_t *msg)
-		PX4_ISFINITE(set_position_target_local_ned.afx) &&
-		PX4_ISFINITE(set_position_target_local_ned.afy) &&
-		PX4_ISFINITE(set_position_target_local_ned.afz) &&
-		PX4_ISFINITE(set_position_target_local_ned.yaw);
+	mavlink_set_position_target_local_ned_t target_local_ned;
+	mavlink_msg_set_position_target_local_ned_decode(msg, &target_local_ned);
 
 	/* Only accept messages which are intended for this system */
-	if ((mavlink_system.sysid == set_position_target_local_ned.target_system ||
-	     set_position_target_local_ned.target_system == 0) &&
-	    (mavlink_system.compid == set_position_target_local_ned.target_component ||
-	     set_position_target_local_ned.target_component == 0) &&
-	    values_finite) {
-
-		offboard_control_mode_s offboard_control_mode{};
-
-		/* convert mavlink type (local, NED) to uORB offboard control struct */
-		offboard_control_mode.ignore_position = (bool)(set_position_target_local_ned.type_mask &
-							(POSITION_TARGET_TYPEMASK_X_IGNORE
-									| POSITION_TARGET_TYPEMASK_Y_IGNORE
-									| POSITION_TARGET_TYPEMASK_Z_IGNORE));
-		offboard_control_mode.ignore_alt_hold = (bool)(set_position_target_local_ned.type_mask &
-							POSITION_TARGET_TYPEMASK_Z_IGNORE);
-		offboard_control_mode.ignore_velocity = (bool)(set_position_target_local_ned.type_mask &
-							(POSITION_TARGET_TYPEMASK_VX_IGNORE
-									| POSITION_TARGET_TYPEMASK_VY_IGNORE
-									| POSITION_TARGET_TYPEMASK_VZ_IGNORE));
-		offboard_control_mode.ignore_acceleration_force = (bool)(set_position_target_local_ned.type_mask &
-				(POSITION_TARGET_TYPEMASK_AX_IGNORE
-				 | POSITION_TARGET_TYPEMASK_AY_IGNORE
-				 | POSITION_TARGET_TYPEMASK_AZ_IGNORE));
-		/* yaw ignore flag mapps to ignore_attitude */
-		offboard_control_mode.ignore_attitude = (bool)(set_position_target_local_ned.type_mask &
-							POSITION_TARGET_TYPEMASK_YAW_IGNORE);
-		offboard_control_mode.ignore_bodyrate_x = (bool)(set_position_target_local_ned.type_mask &
-				POSITION_TARGET_TYPEMASK_YAW_RATE_IGNORE);
-		offboard_control_mode.ignore_bodyrate_y = (bool)(set_position_target_local_ned.type_mask &
-				POSITION_TARGET_TYPEMASK_YAW_RATE_IGNORE);
-		offboard_control_mode.ignore_bodyrate_z = (bool)(set_position_target_local_ned.type_mask &
-				POSITION_TARGET_TYPEMASK_YAW_RATE_IGNORE);
-
-		/* yaw ignore flag mapps to ignore_attitude */
-		bool is_force_sp = (bool)(set_position_target_local_ned.type_mask & POSITION_TARGET_TYPEMASK_FORCE_SET);
-
-		bool is_takeoff_sp = (bool)(set_position_target_local_ned.type_mask & 0x1000);
-		bool is_land_sp = (bool)(set_position_target_local_ned.type_mask & 0x2000);
-		bool is_loiter_sp = (bool)(set_position_target_local_ned.type_mask & 0x3000);
-		bool is_idle_sp = (bool)(set_position_target_local_ned.type_mask & 0x4000);
-		bool is_gliding_sp = (bool)(set_position_target_local_ned.type_mask &
-					    (POSITION_TARGET_TYPEMASK_Z_IGNORE
-					     | POSITION_TARGET_TYPEMASK_VZ_IGNORE
-					     | POSITION_TARGET_TYPEMASK_AZ_IGNORE));
```

Comment:
I guess the gliding setpoint has been removed with this PR

How do we push the systems to go into these behaviors with offboard setpoints? Is this not an option anymore?

Reply:
Thanks, this was seeming a little too easy. I'll review this case.